checks/sbom.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2024 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checks
+
+import (
+	"github.com/ossf/scorecard/v5/checker"
+	"github.com/ossf/scorecard/v5/checks/evaluation"
+	"github.com/ossf/scorecard/v5/checks/raw"
+	sce "github.com/ossf/scorecard/v5/errors"
+	"github.com/ossf/scorecard/v5/probes"
+	"github.com/ossf/scorecard/v5/probes/zrunner"
+)
+
+// SBOM is the registered name for SBOM.
+const CheckSBOM = "SBOM"
+
+//nolint:gochecknoinits
+func init() {
+	supportedRequestTypes := []checker.RequestType{
+		checker.CommitBased,
+		checker.FileBased,
+	}
```

Comment:
I updated this to match webhooks check init function: 

https://github.com/ossf/scorecard/blob/6f9a5122963c8a84474c7db3f45f723ceaa767a1/checks/webhook.go#L34-L39

Since the sbom check wasn't running when I ran against a repo, even with the experimental flag. Is there something that I missed or is this correct?

Reply:
I think no supported request types is the correct registration. 

If this was purely checking source files, then file based and commit based would make sense, but the release aspect means this wouldnt work with `--local` repos, or with a specific commit